UI Issues #1248 and #1247

ShaneWalsh opened the pull request:
#1248 I added a new bounce method to the interface, so when you confirm click on a creature, its queue item not only highlights, but also bounces down once. Cursor also changes to n-resize as requested.

#1247 clicking on a creature now has the same effects as hovering, so the hex's remain after clicking now.

DreadKnight commented:
@ShaneWalsh I've tested this, pretty cool. A few things:
- the n-resize cursor should be displayed since hover, not just during the animation
- the animation shouldn't stack on multiple clicks, because it becomes a bit awkward

It was a bit strange that the status effect under the avatar wasn't pushed down as well, but I guess it's ok.

ktiedt commented:
Manually merged this with my ES6 update -- tested locally, things appear to work as described.

Be sure to pull master down after this is merged, major changes! And thanks @seand88!

ShaneWalsh commented:
I am working on resolving the rest of the issues you raised regards this now. I pulled master and updated my code. 

I think points 1+2 +4+5 should be easy to do.

however "clicking the inactive unit multiple times fast should push the avatar down even more" will be more difficult to do and I may leave that for a separate issue. Will see how I get on.

DreadKnight commented:
@ShaneWalsh Great, was considering merging this and making individual issues, but will give you more time then :)

DreadKnight commented on src/utility/hexagons.js:
```diff
@@ -685,6 +672,25 @@ var HexGrid = class HexGrid {
+		
+		let onCreatureHover = (creature, queueEffect) => {
+			if (creature.type == "--"){
+				if (creature === game.activeCreature){
```
For consistency, there's usually space in between `) {` and comments are capitalized :)

ShaneWalsh commented:
@DreadKnight Please review the latest changes. 
Added a timeout to the bounce effect so it can only occur once a second, regardless of total clicks. When you hover on an inactive creature the cursor changes to n-resize, else its pointer. 
Found this.this bug in code, removed it,
In order to resolve the hidden status on the queue items I had to remove the overflow hidden from the queue css, I hope this doesnt have any adverse affects. I tested and visually it seems fine.
As I said, I have not done "clicking the inactive unit multiple times fast should push the avatar down even more". If you still want it done maybe create another issue to track it.

DreadKnight commented:
@ShaneWalsh Alright, will review soon, hopefully after I manage to get myself a proper new mouse :-/

DreadKnight commented:
@ShaneWalsh Well done, I'll open new issue for the other item. :+1:

DreadKnight commented:
@ShaneWalsh I'll send you an invitation to the organization as a collaborator :)

DreadKnight commented:
@ShaneWalsh I'll send you an invitation to become a collaborator, that way you'll be able to assign issues to yourself. I've opened issue for that other thing #1258 1274
